js/src/logger.ts

```diff
@@ -1065,6 +1102,78 @@ class BackgroundLogger {
     return;
   }
 
+  private registerDroppedItemCount(numItems: number) {
+    if (numItems <= 0) {
+      return;
+    }
+    this.queueDropLoggingState.numDropped += numItems;
+    const timeNow = getCurrentUnixTimestamp();
+    if (
+      timeNow - this.queueDropLoggingState.lastLoggedTimestamp >
+      this.queueDropLoggingPeriod
+    ) {
+      console.warn(
+        `Dropped ${this.queueDropLoggingState.numDropped} elements due to full queue`,
+      );
+      this.queueDropLoggingState.numDropped = 0;
+      this.queueDropLoggingState.lastLoggedTimestamp = timeNow;
+    }
+  }
+
+  private async dumpDroppedEvents(
+    wrappedItems: LazyValue<BackgroundLogEvent>[],
+  ) {
+    const publishPayloadsDir = [
+      this.allPublishPayloadsDir,
+      this.failedPublishPayloadsDir,
+    ].reduce((acc, x) => (x ? acc.concat([x]) : acc), new Array<string>());
+    if (!(wrappedItems.length && publishPayloadsDir.length)) {
+      return;
+    }
+    try {
+      const allItems = await this.unwrapLazyValues(wrappedItems);
+      const dataStr = constructLogs3Data(
+        allItems.map((x) => JSON.stringify(x)),
+      );
+      for (const payloadDir of publishPayloadsDir) {
+        await BackgroundLogger.writePayloadToDir({
```

Comment:
should we log where these are? So if someone sees the message in the logs, they can correlate it to the dropped values in the directory?

Reply:
We could. I guess they are the ones setting the environment variable that tells us to log there, but it doesn't hurt to be more explicit. I guess the problem is to avoid being spammy. So we could add another log-every-n-seconds based message.